server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/SpanServiceElasticsearch.java

```diff
@@ -192,6 +198,61 @@ public void clear(String tenantId) {
         }
     }
 
+    @Override
+    public Trace getTraceFragment(String tenantId, String id) {
+        Span span = getSpan(tenantId, id);
+
+        Trace trace = spanToTrace(span);
+        trace.setNodes(recursiveTraceFragment(tenantId, id));
+
+        return trace;
+    }
+
+    private List<Node> recursiveTraceFragment(String tenantId, String spanId) {
+        List<Span> spanChildren = getChildren(tenantId, spanId);
+        if (spanChildren == null) {
+            return Collections.emptyList();
+        }
+
+        List<Node> nodes = new ArrayList<>();
+
+        for (Span span: spanChildren) {
+            if (span.serverSpan()) {
+                continue;
+            }
+
+            // TODO set uri
+            Consumer consumer = new Consumer(null, "http");
+            consumer.addInteractionId(span.getId());
```

Comment:
The correlation id is correct. However there are other properties on the Consumer (and Component) nodes that will need to be set - including the baseTime and duration, endpointType (for Consumer) and componentType for Component - although this may have to wait until we have more configurable support.